moveit_kinematics/test/test_kinematics_plugin.cpp

```diff
@@ -389,354 +415,339 @@ TEST_F(KinematicsTest, randomWalkIK)
-    EXPECT_NEAR_POSES(poses, reached_poses, tolerance_);
-  }
-
-  ROS_INFO_STREAM("Success Rate: " << (double)success / num_ik_cb_tests_);
-  EXPECT_GE(success, EXPECTED_SUCCESS_RATE * num_ik_cb_tests_);
-}
-
-TEST_F(KinematicsTest, getIK)
-{
-  std::vector<double> fk_values, solution;
-  moveit_msgs::msg::MoveItErrorCodes error_code;
-  solution.resize(kinematics_solver_->getJointNames().size(), 0.0);
-  const std::vector<std::string>& fk_names = kinematics_solver_->getTipFrames();
-  robot_state::RobotState robot_state(robot_model_);
-  robot_state.setToDefaultValues();
-
-  for (unsigned int i = 0; i < num_ik_tests_; ++i)
-  {
-    fk_values.resize(kinematics_solver_->getJointNames().size(), 0.0);
-    robot_state.setToRandomPositions(jmg_, this->rng_);
-    robot_state.copyJointGroupPositions(jmg_, fk_values);
-    std::vector<geometry_msgs::Pose> poses;
-
-    ASSERT_TRUE(kinematics_solver_->getPositionFK(fk_names, fk_values, poses));
-    kinematics_solver_->getPositionIK(poses[0], fk_values, solution, error_code);
-    // starting from the correct solution, should yield the same pose
-    EXPECT_EQ(error_code.val, error_code.SUCCESS);
-
-    Eigen::Map<Eigen::ArrayXd> sol(solution.data(), solution.size());
-    Eigen::Map<Eigen::ArrayXd> truth(fk_values.data(), fk_values.size());
-    EXPECT_TRUE(sol.isApprox(truth, tolerance_)) << sol.transpose() << std::endl << truth.transpose() << std::endl;
-  }
-}
-
-TEST_F(KinematicsTest, getIKMultipleSolutions)
-{
-  std::vector<double> seed, fk_values;
-  std::vector<std::vector<double>> solutions;
-  kinematics::KinematicsQueryOptions options;
-  kinematics::KinematicsResult result;
-
-  const std::vector<std::string>& fk_names = kinematics_solver_->getTipFrames();
-  robot_state::RobotState robot_state(robot_model_);
-  robot_state.setToDefaultValues();
-
-  unsigned int success = 0;
-  for (unsigned int i = 0; i < num_ik_multiple_tests_; ++i)
-  {
-    seed.resize(kinematics_solver_->getJointNames().size(), 0.0);
-    fk_values.resize(kinematics_solver_->getJointNames().size(), 0.0);
-    robot_state.setToRandomPositions(jmg_, this->rng_);
-    robot_state.copyJointGroupPositions(jmg_, fk_values);
-    std::vector<geometry_msgs::Pose> poses;
-    ASSERT_TRUE(kinematics_solver_->getPositionFK(fk_names, fk_values, poses));
-
-    solutions.clear();
-    kinematics_solver_->getPositionIK(poses, fk_values, solutions, result, options);
-
-    if (result.kinematic_error == kinematics::KinematicErrors::OK)
-      success += solutions.empty() ? 0 : 1;
-    else
-      continue;
-
-    std::vector<geometry_msgs::Pose> reached_poses;
-    for (const auto& s : solutions)
-    {
-      kinematics_solver_->getPositionFK(fk_names, s, reached_poses);
-      EXPECT_NEAR_POSES(poses, reached_poses, tolerance_);
-    }
-  }
-
-  ROS_INFO_STREAM("Success Rate: " << (double)success / num_ik_multiple_tests_);
-  EXPECT_GE(success, EXPECTED_SUCCESS_RATE * num_ik_multiple_tests_);
+  // if (publish_trajectory)
```

Comment:
What about all those tests below?

Reply:
They need the launch files port first, we could merge this now and enable them again in the future, right now we have two enabled unit tests

Reply:
ok

Reply:
Please add a TODO explaining this

Reply:
@nbbrooks